marlowe/src/Language/Marlowe/Semantics.hs

```diff
@@ -1,90 +1,134 @@
+import           Ledger.Scripts             (DataScript (..))
+import           Ledger.Validation
+import qualified Prelude                    as P
+
+{-# ANN module ("HLint: ignore Avoid restricted function" :: String) #-}
+
+{- Functions that used in Plutus Core must be inlineable,
+   so their code is available for PlutusTx compiler -}
+{-# INLINABLE accountOwner #-}
+{-# INLINABLE inBounds #-}
+{-# INLINABLE fixInterval #-}
+{-# INLINABLE evalValue #-}
+{-# INLINABLE evalObservation #-}
+{-# INLINABLE refundOne #-}
+{-# INLINABLE moneyInAccount #-}
+{-# INLINABLE updateMoneyInAccount #-}
+{-# INLINABLE addMoneyToAccount #-}
+{-# INLINABLE giveMoney #-}
+{-# INLINABLE reduceContractStep #-}
+{-# INLINABLE reduceContractUntilQuiescent #-}
+{-# INLINABLE applyCases #-}
+{-# INLINABLE applyInput #-}
+{-# INLINABLE convertReduceWarnings #-}
+{-# INLINABLE applyAllInputs #-}
+{-# INLINABLE getSignatures #-}
+{-# INLINABLE checkSignatures #-}
+{-# INLINABLE computeTransaction #-}
+{-# INLINABLE contractLifespanUpperBound #-}
+{-# INLINABLE totalBalance #-}
+{-# INLINABLE validatePayments #-}
+{-# INLINABLE marloweValidator #-}
+
+-- * Aliaces
 
 type Party = PubKey
-type ChoiceName = Text     -- Needs to be updated in playground.
 type NumAccount = Integer
 type Timeout = Slot
 type Money = Ada
+type ChoiceName = ByteString
 type ChosenNum = Integer
+type SlotInterval = (Slot, Slot)
+type TransactionSignatures = Map Party Bool
+
+-- * Data Types
+
+{-| Party account id.
+    Accounts have a number NumAccount and an owner,
+    who is a Party to the contract.
+    @
+    AccountId 0 alicePK
+    AccountId 1 alicePK
+    @
+    Note that alicePK is the owner here in the sense that she will be
+    refunded any money in the account when the contract terminates.
+-}
+data AccountId = AccountId Integer Party
+  deriving stock (Show,Read,Generic,P.Eq,P.Ord)
+  deriving anyclass (Pretty)
 
-data AccountId = AccountId NumAccount Party
-  deriving (Eq,Ord,Show,Read,Generic,Pretty)
 
-accountOwner :: AccountId -> Party
-accountOwner (AccountId _ party) = party
+{-| Choices – of integers – are identified by ChoiceId
+    which combines a name for the choice with the Party who had made the choice.
+-}
+data ChoiceId = ChoiceId ByteString Party
+  deriving stock (Show,Read,Generic,P.Eq,P.Ord)
+  deriving anyclass (Pretty)
 
-data ChoiceId = ChoiceId ChoiceName Party
-  deriving (Eq,Ord,Show,Read,Generic,Pretty)
 
-newtype ValueId = ValueId Text
-  deriving stock (Eq,Ord,Generic)
+{-| Values, as defined using Let are identified by name,
+    and can be used by 'UseValue' construct.
+-}
+newtype ValueId = ValueId ByteString
+  deriving stock (Show,Read,Generic,P.Eq,P.Ord)
+  deriving anyclass (Pretty)
 
-instance Pretty ValueId where
-  prettyFragment = text . show
 
-instance Show ValueId where
-  showsPrec p (ValueId txt) = showsPrec p (T.unpack txt)
-instance Read ValueId where
-  readsPrec p x = [(ValueId (T.pack v), s) | (v, s) <- readsPrec p x]
```

Comment:
I think this will break the communication with Purescript regarding symbolic execution, unless you changed it there as well

Reply:
Or unless deriving stock does some magic I don't know about